extensions/omni_httpc/omni_httpc.c

```diff
@@ -317,6 +317,13 @@ static h2o_httpclient_head_cb on_connect(h2o_httpclient_t *client, const char *e
   *_method = req->method;
   *num_headers = req->num_headers;
   *headers = req->headers;
+    /* 
+    required to prevent malformed host header with unix domain sockets
+    for eg. when calling http://[unix:/var/run/docker.sock]/_ping
+    h2o sets the host header to [unix:/var/run/docker.sock] and golang
+    disallows '/' in host header value with 400 Bad request
+  */
+  req->url.authority.len = 0;
```

Comment:
This is a hack for now, the eventual purpose is to set host header for unix domain socket connections to empty value. Perhaps `http_execute_with_options` could have a new option for indicating the underlying transport(in this case unix domain socket) to configure `h2o` requests` appropriately